### ISSUE TITLE
Missing text content/aria labels on wp-admin components

### ISSUE THREAD

**Issue by joelclimbsthings** [woocommerce/woocommerce-admin#6201]
**Describe the bug**
As part of an accessibility audit, I've noticed a few items within `wc-admin` that are invisible to screen readers due to missing text content, or other labels.

Items:
- Logo in new navigation that toggles the navigation visibility
- Select above charts in analytics reports that allows you to change the period type

**To Reproduce**

1. Use a screen reader
2. Navigate via the keyboard (tab/shift-tab) to the described components
3. Notice that no audio description is given to describe the state.

**Expected behavior**
The content should be provided that allows the screen reader to describe the state/control accurately to the user
